src/data_structures/interval_tree.rs

```diff
@@ -100,6 +100,69 @@ impl<'a, N: Ord + Clone + 'a, D: 'a> Iterator for IntervalTreeIterator<'a, N, D>
     }
 }
 
+/// A `find_mut` query on the interval tree does not directly return references to the nodes in the tree, but
+/// wraps the fields `interval` and `data` in an `EntryMut`. Only the data part can be mutably accessed
+/// using the `data` method
+#[derive(PartialEq, Eq, Debug)]
+pub struct EntryMut<'a, N: Ord + Clone + 'a, D: 'a> {
+    data: &'a mut D,
+    interval: &'a Interval<N>,
+}
+
+impl<'a, N: Ord + Clone + 'a, D: 'a> EntryMut<'a, N, D> {
+    /// Get a mutable reference to the data for this entry
+    pub fn data(&'a mut self) -> &'a mut D {
+        &mut self.data
+    }
+
+    /// Get a reference to the interval for this entry
+    pub fn interval(&self) -> &'a Interval<N> {
```

Comment:
On the road to 100% test coverage, could you additionally call this in your test case?